packages/react-native-session-replay/android/src/main/kotlin/com/datadog/reactnative/sessionreplay/ReactTextShadowNodeUtils.kt

```diff
@@ -0,0 +1,112 @@
+package com.datadog.reactnative.sessionreplay
+
+import android.annotation.SuppressLint
+import android.graphics.Typeface
+import android.util.Log
+import android.widget.TextView
+import androidx.annotation.VisibleForTesting
+import com.facebook.react.bridge.ReactContext
+import com.facebook.react.uimanager.ReactShadowNode
+import com.facebook.react.uimanager.UIImplementation
+import com.facebook.react.uimanager.UIManagerModule
+import com.facebook.react.views.text.ReactTextShadowNode
+import com.facebook.react.views.text.TextAttributes
+import okhttp3.internal.notify
+import okhttp3.internal.wait
+
+internal class ReactTextShadowNodeUtils(
+    private val reactContext: ReactContext
+) {
+    internal fun getFontSize(view: TextView, shadowNode: ReactTextShadowNode?): Long {
+        shadowNode?.javaClass?.superclass?.getDeclaredField("mTextAttributes")?.let {
+            it.isAccessible = true
+            val textAttributes = it.get(shadowNode) as TextAttributes
+            return textAttributes.effectiveFontSize.toLong()
+        }
+
+        return view.textSize.toLong()
+    }
+
+    internal fun getFontFamily(view: TextView, shadowNode: ReactTextShadowNode?): String {
+        shadowNode?.javaClass?.superclass?.getDeclaredField("mFontFamily")?.let {
+            it.isAccessible = true
+            return it.get(shadowNode)?.toString()
+                ?: resolveFontFamily(view.typeface)
+        }
+
+        return resolveFontFamily(view.typeface)
+    }
+
+    internal fun getGravity(view: TextView, shadowNode: ReactTextShadowNode?): Int {
+        shadowNode?.javaClass?.superclass?.getDeclaredField("mTextAlign")?.let {
+            it.isAccessible = true
+            return it.getInt(shadowNode)
+        }
+
+        return view.gravity
+    }
+
+    internal fun getColor(view: TextView, shadowNode: ReactTextShadowNode?): Int {
+        shadowNode?.javaClass?.superclass?.getDeclaredField("mColor")?.let {
+            it.isAccessible = true
+            return it.getInt(shadowNode)
+        }
+
+        return view.currentTextColor
+    }
+
+    internal fun getShadowNode(viewId: Int): ReactShadowNode<out ReactShadowNode<*>>? {
+        val uiManagerModule = try {
+            reactContext.getNativeModule(UIManagerModule::class.java) as UIManagerModule
```

Comment:
I think we should not get the `uiManagerModule` every time we need to get a `shadowNode`, that could have a significant impact on performance if there is a lot of text.

What do you think? I don't think this `UIManagerModule` will change during the lifetime of the app.

We could for instance set the `uiManagerModule` when initializing this helper.